core/src/main/java/jenkins/slaves/EncryptedSlaveAgentJnlpFile.java

```diff
@@ -64,13 +70,13 @@ public EncryptedSlaveAgentJnlpFile(AccessControlled it, String viewName, String
     }
 
     @Override
-    public void generateResponse(StaplerRequest req, StaplerResponse res, Object node) throws IOException, ServletException {
+    public void generateResponse(StaplerRequest req, final StaplerResponse res, Object node) throws IOException, ServletException {
         RequestDispatcher view = req.getView(it, viewName);
         if ("true".equals(req.getParameter("encrypt"))) {
-            final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            final CapturingServletOutputStream csos = new CapturingServletOutputStream();
```

Comment:
Or we could just remove this old hack and directly render the JNLP template into a memory buffer using raw Jelly calls or something, not using a mock response.

Reply:
I went for the conservative approach here as the code was not super easy to understand what it was doing to begin with,  If someone wants to do a better thing then go for it!